server/fcm/fcm_test.go

```diff
@@ -1,468 +1,143 @@
-	// and a http context
-	req := &http.Request{URL: u, Method: "HEAD"}
-	w := httptest.NewRecorder()
-
-	// do a GET instead of POST
-	g.ServeHTTP(w, req)
-
-	// check the result
-	a.Equal("{\"error\":\"method not allowed\"}\n", string(w.Body.Bytes()))
-	a.Equal(w.Code, http.StatusMethodNotAllowed)
-
-	// send a new request with wrong parameters encoding
-	req.Method = "POST"
-	req.URL, _ = u.Parse("http://localhost/fcm/marvin/fcmId123/subscribe3/notifications")
-
-	w2 := httptest.NewRecorder()
-	g.ServeHTTP(w2, req)
-
-	a.Equal("{\"error\":\"invalid parameters in request\"}\n", string(w2.Body.Bytes()))
-	a.Equal(w2.Code, http.StatusBadRequest)
+var fullFCMMessage = `{
+	"notification": {
+		"title": "TEST",
+		"body": "notification body",
+		"icon": "ic_notification_test_icon",
+		"click_action": "estimated_arrival"
+	},
+	"data": {"field1": "value1", "field2": "value2"}
+}`
+
+type mocks struct {
+	router    *MockRouter
+	store     *MockMessageStore
+	gcmSender *MockSender
 }
 
 //TODO Cosmin Bogdan test should be re-enabled after Check() works and is a public func
-//func TestConnector_Check(t *testing.T) {
-//	_, finish := testutil.NewMockCtrl(t)
-//	defer finish()
-//
-//	a := assert.New(t)
-//	gcm, _, _ := testGCMResponse(t, testutil.SuccessGCMResponse)
-//
-//	done := make(chan bool)
-//	mockSender := testutil.CreateGcmSender(
-//		testutil.CreateRoundTripperWithJsonResponse(
-//			http.StatusOK, testutil.SuccessGCMResponse, done))
-//	gcm.Sender = mockSender
-//	err := gcm.check()
-//	a.NoError(err)
-//
-//	done2 := make(chan bool)
-//	mockSender2 := testutil.CreateGcmSender(
-//		testutil.CreateRoundTripperWithJsonResponse(
-//			http.StatusUnauthorized, "", done2))
-//	gcm.Sender = mockSender2
-//	a.Error(gcm.check())
-//}
-
-func TestFCM_SaveAndLoadSubs(t *testing.T) {
-	_, finish := testutil.NewMockCtrl(t)
-	defer finish()
-
-	a := assert.New(t)
-	g, routerMock, _ := testFCMResponse(t, testutil.SuccessFCMResponse)
-
-	// given: some test routes
-	testRoutes := map[string]bool{
-		"marvin:/foo:1234": true,
-		"zappod:/bar:1212": true,
-		"athur:/erde:42":   true,
-	}
-
-	routerMock.EXPECT().Subscribe(gomock.Any()).Do(func(route *router.Route) {
-		// delete the route from the map, if we got it in the test
-		delete(testRoutes, fmt.Sprintf("%v:%v:%v", route.Get(userIDKey), route.Path, route.Get(applicationIDKey)))
-	}).AnyTimes()
-
-	// when: we save the routes
-	for k := range testRoutes {
-		splitKey := strings.SplitN(k, ":", 3)
-		userID := splitKey[0]
-		topic := splitKey[1]
-		fcmID := splitKey[2]
-		initSubscription(g, topic, userID, fcmID, 0, true)
-	}
-
-	// and reload the routes
-	g.loadSubscriptions()
-
-	time.Sleep(50 * time.Millisecond)
-
-	// then: all expected subscriptions were called
-	a.Equal(0, len(testRoutes))
-}
-
-func TestRemoveTrailingSlash(t *testing.T) {
-	assert.Equal(t, "/foo", removeTrailingSlash("/foo/"))
-	assert.Equal(t, "/foo", removeTrailingSlash("/foo"))
-}
+// func TestConnector_Check(t *testing.T) {
```

Comment:
this should be un-commented if possible, and the TODO removed
now the Check func is public again, needs a test
if Check is not really working now, there is a task I created earlier on JIRA -> healthchecks for apns+fcm